server/src/main/java/org/opensearch/search/profile/aggregation/ConcurrentAggregationProfiler.java

```diff
@@ -0,0 +1,216 @@
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.search.profile.aggregation;
+
+import org.opensearch.search.profile.ProfileResult;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Main class to profile aggregations with concurrent execution
+ *
+ * @opensearch.internal
+ */
+public class ConcurrentAggregationProfiler extends AggregationProfiler {
+
+    private static final String MAX_PREFIX = "max_";
+    private static final String MIN_PREFIX = "min_";
+    private static final String AVG_PREFIX = "avg_";
+    private final String[] breakdownCountStatsTypes = { "build_leaf_collector_count", "collect_count" };
+
+    @Override
+    public List<ProfileResult> getTree() {
+        List<ProfileResult> tree = profileTree.getTree();
+        List<ProfileResult> reducedTree = new LinkedList<>();
+        Map<String, List<ProfileResult>> sliceLevelAggregationMap = getSliceLevelAggregationMap(tree);
+        for (List<ProfileResult> profileResultsAcrossSlices : sliceLevelAggregationMap.values()) {
+            reducedTree.addAll(reduceProfileResultsTree(profileResultsAcrossSlices));
+        }
+        return reducedTree;
+    }
+
+    private List<ProfileResult> reduceProfileResultsTree(List<ProfileResult> profileResultsAcrossSlices) {
+        String type = profileResultsAcrossSlices.get(0).getQueryName();
+        String description = profileResultsAcrossSlices.get(0).getLuceneDescription();
+        long maxSliceNodeEndTime = Long.MIN_VALUE;
+        long minSliceNodeStartTime = Long.MAX_VALUE;
+        long maxSliceNodeTime = Long.MIN_VALUE;
+        long minSliceNodeTime = Long.MAX_VALUE;
+        long avgSliceNodeTime = 0L;
+        Map<String, Long> breakdown = new HashMap<>();
+        Map<String, Long> timeStatsMap = new HashMap<>();
+        Map<String, Long> minSliceStartTimeMap = new HashMap<>();
+        Map<String, Long> maxSliceEndTimeMap = new HashMap<>();
+        Map<String, Long> countStatsMap = new HashMap<>();
+        Map<String, Object> debug = new HashMap<>();
+        List<ProfileResult> children = new LinkedList<>();
+
+        for (ProfileResult profileResult : profileResultsAcrossSlices) {
+            long profileNodeTime = profileResult.getTime();
+            long sliceStartTime = profileResult.getTimeBreakdown().get("initialize_startTime");
```

Comment:
We can define a static constant like 
`private final static String START_TIME_KEY = AggregationTimingType.INITIALIZE.toString() + TIMINGTYPE_START_TIME_SUFFIX` (defined above in AbstractProfileBreakdown)

Reply:
It has been added now. Thank you.